.github/workflows/test-upgrade.yaml

```
@@ -0,0 +1,134 @@
+---
+name: Kuadrant Upgrade Test
+
+on:
+  workflow_dispatch:
+    inputs:
+      kuadrantStartVersion:
+        description: Kuadrant start version
+        required: true
+        type: string
+      clusterServer:
+        description: Cluster server URL
+        required: false
+        type: string
+      clusterToken:
+        description: Cluster Server Bearer Token
+        required: false
+        type: string
+      kuadrantNamespace:
+        description: Namespace where Kuadrant is installed
+        required: false
+        default: kuadrant-system
+        type: string
+jobs:
+  helm-charts-upgrade-test:
+    runs-on: ubuntu-latest
+    name: Helm Charts Upgrade Test
+    env:
+      KIND_CLUSTER_NAME: kuadrant-test
+      K8S_USER: kuadrant-admin  # can be whatever, it does not matter.
+      CLUSTER_NAME: remote-cluster
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+      - name: Install helm
+        run: |
+          make helm
+      - name: Install yq tool
+        run: |
+          # following sub-shells running make target should have yq already installed
+          make yq
+      - name: Determine upgrade version
+        id: upgrade-version
+        run: |
+          version=`make helm-print-chart-version`
+          echo version=$version >> $GITHUB_OUTPUT
+      - name: Print versions
+        run: echo "Installing version ${{ inputs.kuadrantStartVersion }}, upgrading to version ${{ steps.upgrade-version.outputs.version }}"
+      - name: Deploy local Kind cluster
+        if: ${{ github.event_name == 'workflow_dispatch' && inputs.clusterServer == '' }}
+        uses: helm/kind-action@v1.12.0
+        with:
+          version: v0.23.0
+          config: utils/kind-cluster.yaml
+          cluster_name: ${{ env.KIND_CLUSTER_NAME }}
+          wait: 120s
+      - name: Install kubectl for remote cluster
+        if: ${{ github.event_name == 'workflow_dispatch' && inputs.clusterServer != '' }}
```

What we could do is having a job scoped env: 

```yaml
env:
  LOCAL_CLUSTER: ${{ inputs.clusterServer == '' && inputs.clusterToken != ''}}
```

thus in checks like this, we could do:

```suggestion
        if: ${{ !env.LOCAL_CLUSTER }}
```

also, the `github.event_name == 'workflow_dispatch'` is not needed in the conditional

wdyt?

---

doing that, yes